packages/loot-core/src/server/db/index.ts

```diff
@@ -623,14 +618,69 @@ export async function getTransactions(accountId) {
   );
 }
 
-export function insertTransaction(transaction) {
+export async function insertTransaction(transaction) {
+  const lastTransaction = await first(
+    'SELECT sort_order FROM v_transactions ' +
+      'WHERE date = ? AND is_child <> 1 AND parent_id IS NULL ' +
+      'ORDER BY sort_order DESC LIMIT 1',
+    [transaction.date.replace(/-/g, '')], // Remove hyphens
+  );
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏this could potentially be a pretty expensive SQL query. Especially for bigger databases.

Do we _really_ need the last transactions `sort_order` when creating a new transaction? Could we just set it to `99999` or something like that? And then change sorting from `sort_by:sort_order` to `sort_by:sort_order,insertion_id` (i.e. two index sorting).

WDYT?